met-web/src/components/metadataManagement/TaxonEditForm.tsx

```diff
@@ -0,0 +1,325 @@
+            preset_values: taxon.preset_values ?? [],
+        });
+    }, [taxon, reset]);
+
+    // Watch data_type to update taxonType
+    const dataType = watch('data_type');
+    const taxonType = TaxonTypes[dataType as keyof typeof TaxonTypes];
+    // Watch freeform to update label
+    const isFreeform = watch('freeform');
+    const isMulti = !watch('one_per_engagement');
+    const presetValues = watch('preset_values');
+
+    const schema = yup.object().shape({
+        name: yup.string().required('Name is required').max(64, 'Name is too long!'),
+        description: yup.string().max(255, 'Description is too long!').strip(),
+        freeform: yup.boolean().oneOf([taxonType.supportsFreeform, false]),
+        one_per_engagement: taxonType.supportsMulti ? yup.boolean() : yup.boolean().oneOf([true]),
+        data_type: yup.string().required('Type is required'),
+        preset_values: yup.mixed().when('data_type', {
+            is: (value: string) => TaxonTypes[value as keyof typeof TaxonTypes].supportsPresetValues,
+            then: yup.mixed().when('freeform', {
+                is: false,
+                then: yup
+                    .array()
+                    .of(taxonType.yupValidator ?? yup.mixed())
+                    .required('Preset value is required'),
+                otherwise: yup.array().of(taxonType.yupValidator ?? yup.mixed()),
+            }),
+            otherwise: yup.mixed().strip(),
+        }),
+    });
+
+    const onSubmit: SubmitHandler<MetadataTaxon> = async (data, event) => {
+        let formErrors: { [key: string]: string } = {};
+        // These fields don't always apply to all taxon types, so
+        // if they are unsupported, we set them to the defaults
+        if (!taxonType.supportsFreeform) data.freeform = false;
+        if (!taxonType.supportsMulti) data.one_per_engagement = true;
+        if (!taxonType.supportsPresetValues) data.preset_values = [];
+        try {
+            await schema.validate(data, { abortEarly: false });
+            // Catch clause variable type annotation must be 'any' or 'unknown' if specified
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: yup.ValidationError | any) {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            formErrors = error.inner.reduce((errors: { [key: string]: string }, innerError: any) => {
+                errors[innerError.path] = innerError.message;
+                return errors;
+            }, {});
+            // Set errors for each field in formState
+            Object.keys(formErrors).forEach((fieldName) => {
+                setError(fieldName as keyof MetadataTaxon, {
+                    type: 'validate',
+                    message: formErrors[fieldName],
+                });
+            });
+        }
+
+        if (Object.keys(formErrors).length) {
+            console.log('Form errors:', formErrors);
```

Comment:
My personal preference would be to only have console logs for errors in logic, network, or other programming concerns. I feel like the validation errors could be concerned an expected part of application behaviour